core/templates/dev/head/pages/topics_and_skills_dashboard/TopicsAndSkillsDashboard.js

```diff
@@ -0,0 +1,43 @@
+// Copyright 2018 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Controllers for the topics and skills dashboard.
+ */
+oppia.controller('TopicsAndSkillsDashboard', [
+  '$scope', '$rootScope', '$http', '$window',
+  'AlertsService', 'TopicsAndSkillsDashboardBackendApiService',
+  'UrlInterpolationService', 'FATAL_ERROR_CODES',
+  function(
+      $scope, $rootScope, $http, $window,
+      AlertsService, TopicsAndSkillsDashboardBackendApiService,
+      UrlInterpolationService, FATAL_ERROR_CODES) {
+    TopicsAndSkillsDashboardBackendApiService.fetchDashboardData().then(
+      function(response) {
+        $scope.topicSummaries = response.data.topic_summaries;
+        $scope.skillSummaries = response.data.skill_summaries;
+        $scope.activeTab = 'topics';
+      },
+      function(errorResponse) {
+        if (FATAL_ERROR_CODES.indexOf(errorResponse.status) !== -1) {
+          AlertsService.addWarning('Failed to get dashboard data');
+        }
```

Comment:
else error somehow?

Reply:
FATAL_ERROR_CODES has 400, 401, 404 and 500. So, would an else be needed? or shall I remove the if condition entirely?

Reply:
Could remove the if condition, or do something like `else "unexpected error code from server"`. Probably want to be noisy somehow if that sort of error shouldn't happen in the first place, I guess.

Reply:
Done, added an else statement.